aziotctl/src/internal/check/checks/well_formed_configs.rs

```diff
@@ -171,14 +189,71 @@ impl Checker for WellFormedIdentitydConfig {
 enum DaemonCfg<T> {
     Cfg(T),
     PermissionDenied(Error),
+    IncorrectPermissions(Error),
+    IncorrectGroupId(Error),
+    IncorrectUserID(Error),
 }
 
 async fn load_daemon_cfg<T: serde::de::DeserializeOwned>(
     daemon: &str,
+    uid: &str,
     config_path: &Path,
     config_directory_path: Option<&Path>,
     shared: &CheckerShared,
 ) -> Result<DaemonCfg<T>> {
+    use std::os::linux::fs::MetadataExt;
+    if let Ok(metadata) = fs::metadata(config_path) {
+        if !metadata.permissions().readonly() {
+            return Ok(DaemonCfg::IncorrectPermissions(anyhow::anyhow!(
+                "The file {} must be read only. Please run 'sudo chmod 0444 {}'",
+                config_path.to_string_lossy(),
+                config_path.to_string_lossy()
+            )));
+        }
+
+        let user = match User::from_uid(Uid::from_raw(metadata.st_uid())) {
+            Ok(Some(user)) => user.name,
+            Err(_) | Ok(None) => {
+                return Err(anyhow::anyhow!(format!(
+                    "Could not find user assigned to {}",
+                    config_path.to_string_lossy()
+                )))
+            }
+        };
+
+        let group = match Group::from_gid(Gid::from_raw(metadata.st_gid())) {
+            Ok(Some(group)) => group.name,
+            Err(_) | Ok(None) => {
+                return Err(anyhow::anyhow!(format!(
+                    "Could not find user group assigned to {}.",
+                    config_path.to_string_lossy()
+                )))
+            }
+        };
+
+        if !user.eq(uid) {
+            return Ok(DaemonCfg::IncorrectUserID(anyhow::anyhow!(
+                "The file {} has user {}, expected {}. Please run 'sudo chown {} {}'",
+                config_path.to_string_lossy(),
+                user,
+                uid,
+                uid,
+                config_path.to_string_lossy()
+            )));
+        }
+
+        if !group.eq(uid) {
+            return Ok(DaemonCfg::IncorrectGroupId(anyhow::anyhow!(
+                "The file {} has group {}, expected {}. Please run 'sudo chgrp {} {}' ",
+                config_path.to_string_lossy(),
+                group,
+                uid,
+                uid,
+                config_path.to_string_lossy()
+            )));
+        }
+    }
+
```

Comment:
Do the same thing as: https://github.com/Azure/iot-identity-service/blob/18f5e5c92a68d08f37ddafeceb16075f1260bcb7/aziotctl/src/config/apply.rs#L41-L69 ie if the current user is root then do an actual check. Otherwise if this is a debug build, allow the file to be owned by the current user.